std/concurrency.d

```diff
@@ -1076,6 +1076,33 @@ Tid locate( string name )
     }
 }
 
+/**
+ * Gets an array all of the names associated with the $(LREF Tid).
+ *
+ * Params:
+ *     tid = The $(LREF Tid) to locate within the registry.
+ *
+ * Returns:
+ *     An array of associated names, or `null` if `tid` is not registered.
+ *
+ * See_Also:
+ *     $(LREF register)
+ */
+string[] locate(Tid tid)
+{
+    synchronized (registryLock)
+    {
+        return namesByTid.get(tid, null);
+    }
+}
+
+///
+unittest
+{
+    auto id = thisTid();
+    assert(register("Main", id));
+    assert(locate(id) == ["Main"]);
```

Comment:
This will fail if another unittest has registered anything for the main thread. If you think that matters, you could wrap the whole unittest in `spawn` so you know you'll have a new `id`.